cookbooks/aws-parallelcluster-config/recipes/enable_chef_error_handler.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster
+# Recipe:: enable_chef_error_handler
+#
+# Copyright:: 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+if node["cluster"]["node_type"] == "HeadNode"
+  chef_handler 'WriteChefError::WriteChefError' do
+    type exception: true
+    action :enable
+  end
+end
+
+include_recipe "aws-parallelcluster-slurm::enable_chef_error_handler" if node["cluster"]["scheduler"] == "slurm"
```

Comment:
What about 
rename the WriteChefError to WriteHeadNodeChefError

same for emit_chef_error_events --> WriteHComputeFleetChefError

What about:
And remove if node["cluster"]["node_type"] == "ComputeFleet" in slurm recipe?
```
if node["cluster"]["node_type"] == "HeadNode"
  chef_handler 'WriteChefError::WriteHeadNodeChefError' do
    type exception: true
    action :enable
  end
elsif node["cluster"]["scheduler"] == "slurm" && node["cluster"]["node_type"] == "ComputeFleet"
  include_recipe "aws-parallelcluster-slurm::enable_chef_error_handler"
end
```

Reply:
I agree with changing names of these handlers.
`WriteChefError::WriteChefError` -> `WriteChefError::WriteHeadNodeChefError`
`WriteChefError::SlurmChefHandler` -> `WriteChefError::WriteComputeFleetSlurmChefError`

In my opinion:
- Although now we only have one slurm handler (the compute node handler), maybe in the future we can have another slurm handler, regardless of node type
- In this case, instead of hard code it in config (checking the node type), we may want to let slurm to make the decision on when and which slurm handler(s) should be enabled 
- This `aws-parallelcluster-slurm::enable_chef_error_handler` recipe will do exactly this job, so we should have the condition check logic within this recipe